src/main/java/org/embulk/output/s3_per_record/S3PerRecordOutputPlugin.java

```diff
@@ -202,7 +203,13 @@ public void add(Page page) {
                            throw e;
 
                         retryCount++;
-                        logger.warn(String.format("> Upload failed by %s, Retry Uploading (%d of %d)", e.getMessage(), retryCount, retryLimit));
+                        logger.warn(String.format("> Upload failed by %s, Retry Uploading in after %d ms (%d of %d)", e.getMessage(), retryWait, retryCount, retryLimit));
+                        try {
+                            Thread.sleep(retryWait);
+                            retryWait = retryWait * 2;
+                        } catch (InterruptedException e1) {
+                            throw new RuntimeException(e1);
```

Comment:
IMO, it is too much to raise exception for interrupts. `Thread.currentThread().interrupt();` will be enough.
http://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#2.1 discusses this topic.